src/history_manager.py

```diff
@@ -5,6 +5,36 @@
 
 ################################################################################
 
+# 
+
+class State():
```

Comment:
could you name it DrHistoryState and move it to a file named `history_state.py` please? my naming conventions are probably terrible but i try to follow them consistently anyway lol

then you'll need to add a line in `src/meson.build` so the class will be importable in the history manager